lib/components/base/Pagination.vue

```diff
@@ -49,66 +49,64 @@
     </a>
   </div>
 </template>
-<script setup>
+<script setup lang="ts">
+import { computed } from 'vue'
 import { GapIcon, LeftArrowIcon, RightArrowIcon } from '@'
-</script>
-<script>
-import { defineComponent } from 'vue'
-
-export default defineComponent({
-  props: {
-    page: {
-      type: Number,
-      default: 1,
-    },
-    count: {
-      type: Number,
-      default: 1,
-    },
-    linkFunction: {
-      type: Function,
-      default() {
-        return null
-      },
-    },
+
+const emit = defineEmits<{
+  'switch-page': [page: number]
+}>()
+
+const props = defineProps({
+  page: {
+    type: Number,
+    default: 1,
   },
-  emits: ['switch-page'],
-  computed: {
-    pages() {
-      let pages = []
-
-      if (this.count > 7) {
-        if (this.page + 3 >= this.count) {
-          pages = [
-            1,
-            '-',
-            this.count - 4,
-            this.count - 3,
-            this.count - 2,
-            this.count - 1,
-            this.count,
-          ]
-        } else if (this.page > 5) {
-          pages = [1, '-', this.page - 1, this.page, this.page + 1, '-', this.count]
-        } else {
-          pages = [1, 2, 3, 4, 5, '-', this.count]
-        }
-      } else {
-        pages = Array.from({ length: this.count }, (_, i) => i + 1)
-      }
-
-      return pages
-    },
+  count: {
+    type: Number,
+    default: 1,
   },
-  methods: {
-    switchPage(newPage) {
-      this.$emit('switch-page', newPage)
-      if (newPage !== null && newPage !== '' && !isNaN(newPage)) {
-        this.$emit('switch-page', Math.min(Math.max(newPage, 1), this.count))
-      }
+  linkFunction: {
+    type: Function,
+    default() {
+      return null
     },
   },
 })
+
+const pages = computed(() => {
+  let pages = []
```

Comment:
```suggestion
  let pages: ('-' | number)[] = []
```

If you narrow the type definition here, you will be able to simplify some checks and avoid type casts.

Reply:
Done ^^.